Docs: fixes 404s pulled from GSC

ZPain8464 opened the pull request:
Should fix: 
- [ ] https://github.com/pomerium/documentation/issues/852
- [ ] https://github.com/pomerium/documentation/issues/851
- [ ] https://github.com/pomerium/documentation/issues/850
- [ ] https://github.com/pomerium/documentation/issues/849
- [ ] https://github.com/pomerium/documentation/issues/848
- [ ] https://github.com/pomerium/documentation/issues/847
- [ ] https://github.com/pomerium/documentation/issues/846
- [ ] https://github.com/pomerium/documentation/issues/845
- [ ] https://github.com/pomerium/documentation/issues/844
- [ ] https://github.com/pomerium/documentation/issues/843
- [ ] https://github.com/pomerium/documentation/issues/842
- [ ] https://github.com/pomerium/documentation/issues/841
- [ ] https://github.com/pomerium/documentation/issues/840
- [ ] https://github.com/pomerium/documentation/issues/839
- [ ] https://github.com/pomerium/documentation/issues/838

ZPain8464 commented:
@kenjenkins yeah, we can add it to this PR. I usually check GSC and just add issues for each broken link; this one wasn't flagged yet, but I'll include it anyway.

kenjenkins commented on static/_redirects:
```diff
@@ -322,6 +324,7 @@ https://0-20-0.docs.pomerium.com/category/guides https://0-20-0.docs.pomerium.co
 # Autocert
 /docs/reference/autocert/autocert-ca /docs/reference/autocert
```
It looks like some redirects include an ID fragment in the URL. I think maybe we should do that for all redirects that now link to a consolidated settings page (e.g. this one would be https://www.pomerium.com/docs/reference/autocert#autocert-ca).

(if we do this, let's do it in a separate PR)

kenjenkins commented on static/_redirects:
```diff
@@ -375,6 +379,21 @@ https://0-20-0.docs.pomerium.com/category/guides https://0-20-0.docs.pomerium.co
+# Metrics
+/docs/reference/metrics-client-certificate-authority /docs/reference/metrics
+
+# gRPC
```
There's also https://www.pomerium.com/docs/reference/grpc-client-timeout, I think.

kenjenkins commented on static/_redirects:
```diff
@@ -375,6 +379,21 @@ https://0-20-0.docs.pomerium.com/category/guides https://0-20-0.docs.pomerium.co
 /docs/reference/branding/primary-color /docs/reference/branding
 /docs/reference/branding/secondary-color /docs/reference/branding
 
+# TLS
```
I think there are still many other redirects to add here, based on https://github.com/pomerium/documentation/commit/e665ce64b11508ae54bf8e8fb33b0596254608e9:

- /docs/reference/routes/tls-client-certificate
- /docs/reference/routes/tls-custom-certificate-authority
- /docs/reference/routes/tls-downstream-client-certificate-authority
- /docs/reference/routes/tls-downstream-server-name
- /docs/reference/routes/tls-skip-verification
- /docs/reference/routes/tls-upstream-allow-renegotiation

kenjenkins commented on static/_redirects:
```diff
@@ -375,6 +379,21 @@ https://0-20-0.docs.pomerium.com/category/guides https://0-20-0.docs.pomerium.co
+/docs/reference/routes/tls-server-name /docs/reference/routes/tls
+/docs/reference/routes/tls-upstream-server-name /docs/reference/routes/tls
+
+# Metrics
```
I think there are three more redirects that should be added here, based on https://github.com/pomerium/documentation/commit/ae81e48ef0cbaccdc9dd657d46b3322294efe963.

ZPain8464 commented:
@kenjenkins I went ahead and added the other consolidated links to this PR. I also added the fragment IDs to each link. Please take a look to review whenever you have a second; thanks!

kenjenkins commented on static/_redirects:
```diff
@@ -320,66 +322,109 @@ https://0-20-0.docs.pomerium.com/category/guides https://0-20-0.docs.pomerium.co
+
+# TLS
+/docs/reference/routes/tls-server-name /docs/reference/routes/tls#tls-upstream-server-name
+/docs/reference/routes/tls-upstream-server-name /docs/reference/routes/tls#tls-upstream-server-name
+/docs/reference/tls-downstream-server-name /docs/reference/routes/tls#tls-downstream-server-name
```
I think we should keep both of the existing redirects from lines 381-382, so something like this?

```suggestion
/docs/reference/routes/tls-upstream-server-name /docs/reference/routes/tls#tls-upstream-server-name
/docs/reference/tls-upstream-server-name /docs/reference/routes/tls#tls-upstream-server-name
/docs/reference/routes/tls-downstream-server-name /docs/reference/routes/tls#tls-downstream-server-name
/docs/reference/tls-downstream-server-name /docs/reference/routes/tls#tls-downstream-server-name
```

kenjenkins reviewed (approved):
Looks good, thanks for doing this!
